dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -3942,15 +3961,14 @@ ReturnedColumn* buildFunctionColumn(Item_func* ifp, gp_walk_info& gwi, bool& non
         nonSupport = true;
         gwi.fatalParseError = true;
         Message::Args args;
-        string info = funcName + " with argument count > " + std::to_string(std::numeric_limits<uint16_t>::max());
+        string info =
+            funcName + " with argument count > " + std::to_string(std::numeric_limits<uint16_t>::max());
         args.add(info);
         gwi.parseErrorText = IDBErrorInfo::instance()->errorMsg(ERR_NON_SUPPORTED_FUNCTION, args);
         return NULL;
       }
-
-      if (!selectBetweenIn && (ifp->arguments()[0]->type() == Item::FIELD_ITEM ||
-                               (ifp->arguments()[0]->type() == Item::REF_ITEM &&
-                                (*(((Item_ref*)ifp->arguments()[0])->ref))->type() == Item::FIELD_ITEM)))
+      Item* ifpArg0 = directRefOrRefLoop(ifp->arguments()[0]);
+      if (!selectBetweenIn && ifpArg0->type() == Item::FIELD_ITEM)
```

Comment:
Again, I am not clear as to why we have a call to `directRefOrRefLoop()` here.

Reply:
The original if condition checks if the ifp::arguments()[0] is Item_field or it is a Item_field behind a ref and directRefOrRefLoop loops till the first non-REF Item is found. directRefOrRefLoop can even return an item itself.